sql-plugin/src/main/spark321db/scala/org/apache/spark/sql/execution/datasources/v2/rapids/GpuAtomicCreateTableAsSelectExec.scala

```diff
@@ -50,15 +49,17 @@ import org.apache.spark.sql.vectorized.ColumnarBatch
  * write fails, the table is instructed to roll back all staged changes.
  */
 case class GpuAtomicCreateTableAsSelectExec(
-    override val output: Seq[Attribute],
+    cpuExec: AtomicCreateTableAsSelectExec,
```

Comment:
We should not be passing the original cpuExec as an argument here. In general we don't want to propagate a CPU node and all of its children as the plan node gets copied around during optimizations. It also makes the explain text messy since the arguments are itself another case class.

Better fix is to remove the AtomicCreate and AtomicReplace code from the common Databricks provider code and move them to overrides in the Databricks--version-specific provider that derives from the abstract common provider, similar to what was done for open-source Delta Lake providers.